### PR TITLE
Return vs print exercise

### PR DESCRIPTION
**Pull request by tlvdberg**
I added an exercise making clear the difference between return and print. As asked for in issue:
https://github.com/swcarpentry/python-novice-inflammation/issues/506

Lines are approximately 807-850.

This is part of an check-out exercise (if accepted).

### CODE REVIEW

**Comment by maxim-belkin** [comment]
Thanks, @tlvdberg! A couple of thoughts:

1. We can discuss `print vs. return` after introducing functions, which we do in episode 6. So, I think this should be moved to episode 6.
2. It might make sense to have a simpler example, something like `print(1)` vs `return 1`  or simply discuss the outputs of `print(print(1))`  or  `print(print(1) + 2)`

I should also note that we should probably revisit the use of the word `"return"` in episodes prior to 6.

**Comment by jttkim** [comment]
I very much like the idea of this exercise. Realising that that which we see and that which the computer "sees", or rather processes are two very different things is a key step in learning to program, and to work with computers in general. It may further be useful to point out parallels to shell scripting, in particular the joint good practice of dealing with errors by writing a meaningful error via `echo ... >& 2` to the standard error and exiting with a non-zero code.

I think @maxim-belkin is right in suggesting to move the exercise to a stage where functions are introduced.

I'd prefer minimising any discussion and dwelling on the `None` value implicitly generated by Python when nothing is returned. The conflation of returning `None` with returning nothing is quite an infelicity of Python's design. Guiding learners to using `type` for exploring returned values is a generally useful practice, but I'm not sure whether they'd get much use or benefit from the `print(print(1))` construct.

**Comment by maxim-belkin** [comment]
I'd avoid drawing parallelism with shell scripting at this point for a few reasons:
1. in shells values are passed by "printing" them and capturing output, rather than by returning values
2. in shells `if success then something fi` is equivalent to `if 0 then something fi` whereas in Python it is equivalent to `if not 0: something`

I had a paragraph on `None` in my original response to this PR but then scrapped it to give myself time to think about it. We use it in lesson 06-func but we don't "explain" it. And I think that any example demonstrating the difference between print and return will inevitably mention `None`

**Comment by maxim-belkin** [comment]
@tlvdberg have a look at swcarpentry/python-novice-inflammation#489

**Comment by tlvdberg** [comment]
Hi All,
I actually also think it is better to discuss this right after introducing function, but as you asked for an exercise, I put it in the exercises. Now, didn't I change this to episode 6 (functions)??? Under 'files changed' it says '_episodes/06-func.md'... What are you referring to?

The problem I would rise is that the lesson is already quite long... I actually think that we could take out the 'Integer Division' section, that is in episode 6 right after introducing functions. Because it's not a problem in Python 3 and discussing things of Python 2 might only confuse learners. It is information they don't need for working successfully with Python 3. Maybe we could replace this by some comments about return vs. print. I would make the lesson much longer than it already is...

About #489: interesting. Maybe good indeed comparing functions with and without return statements.

**Comment by maxim-belkin** [comment]
> Now, didn't I change this to episode 6 (functions)??? Under 'files changed' it says '_episodes/06-func.md'... What are you referring to?

My apologies. Not sure what went wrong here.

> I actually think that we could take out the 'Integer Division' section, that is in episode 6 right after introducing functions.

Let's do that. In fact, we can move it to the "additional exercises" page (`_extras/extra_exercises.md`).

File: _episodes/06-func.md [hunk 1]
```diff
@@ -804,6 +804,51 @@ readable code!
 > {: .solution}
 {: .challenge}
 
+> ## Return versus print
+>
+> Note that `return` and `print` are not interchangeable.
+> `print` is a build-in function, it sends output to the screen.
```
Comment (maxim-belkin) [inline _episodes/06-func.md]: Say:

> `print` is a function that _prints_ data to the screen.

File: _episodes/06-func.md [hunk 2]
```diff
@@ -804,6 +804,51 @@ readable code!
...
+> This allows the *user* to see the data.
+> `return` controls the flow of data between different functions.
```
Comment (maxim-belkin) [inline _episodes/06-func.md]: > `return` allows data to be seen by the program.

File: _episodes/06-func.md [hunk 3]
```diff
@@ -804,6 +804,51 @@ readable code!
...
+> It allows the *program* to move data around.
```
Comment (maxim-belkin) [inline _episodes/06-func.md]: I'd remove this line

File: _episodes/06-func.md [hunk 4]
```diff
@@ -804,6 +804,51 @@ readable code!
...
+> Think about what the following code will give you?
+>
+> ~~~
+> def addto(a,b):
+>     print(a+b)
+> 
+> addto(2,addto(3,4))
+> ~~~
```
Comment (maxim-belkin) [inline _episodes/06-func.md]: Let's, instead, do something like `A = add(1, 2)`

**Comment by maxim-belkin** [comment]
@tlvdberg please [rebase this PR](https://github.com/swcarpentry/python-novice-inflammation/wiki/Rebasing-a-PR)

**Comment by tlvdberg** [comment]
Hi Maxim,
Thanks. I applied the requested changes, and then rebased as you asked. This is the first time I did this, I hope I did it correctly.
I did not more the 'Integer Division' section to the extra exercises section, I guess that could be done in another PR. 
 Let me know if anything's wrong, and I'll correct it...

**Comment by maxim-belkin** [comment]
I've made some changes and merged your exercise. Please have a closer look at it when you have a chance and feel free to submit PRs with further improvements!
Thank you, @tlvdberg!

File: _episodes/06-func.md (hunk 1 above)
Comment (kvndrs) [reply _episodes/06-func.md]: Next, would you indicate that 'return' is not like print in that it does not send output to the screen?

**Pull request merged**